Have Mender to be systemd daemon

pasinskim opened the pull request.

pasinskim commented:
@kacf knowing your prior experience with systemd I would like you to take a look at this.

kacf commented on recipes-mender/mender/files/mender.timer:
```diff
@@ -0,0 +1,11 @@
+Unit=mender.service
+
+[Install]
+WantedBy=multi.user.target
```
As discussed I think we definitely don't want this for Mender. Mender should either be a daemon or launched from some sort of on demand service daemon, such as xinetd (I think systemd also has some hooks for this). But most likely a standalone daemon.

pasinskim replied:
I agree. Fixed!

kacf commented on recipes-mender/mender/files/mender.service:
```diff
@@ -0,0 +1,10 @@
+Type=simple
+User=root
+Group=root
+ExecStart=/usr/bin/mender
```
I envision we might call mender with something like `--daemon` here. But at the moment we don't have it.

pasinskim replied:
Yup. Since we don't have it right now systemd will just fail starting Mender but this is enough for now to continue with Mender Go implementation.

kacf commented on recipes-mender/mender/mender_0.1.bb:
```diff
@@ -7,12 +7,25 @@ S = "${WORKDIR}/git"
+inherit systemd
+
+#FILESEXTRAPATHS_prepend := " ${THISDIR}/mender:"
+#SRC_URI += "file://mender.service"
```
Remove?

pasinskim replied:
Fixed!

kacf replied:
This is not fixed, it's still there!

kacf commented:
Looks good otherwise!

pasinskim commented:
@kacf can you take a second look. I've removed timer that was supposed to run Mender every 5 minutes. I've also added network support so that we will be able to proceed with user-space go app development.

kacf commented on recipes-core/systemd/systemd_%.bbappend:
```diff
@@ -0,0 +1,17 @@
+        install -d ${D}${sysconfdir}/systemd/network
+        install -m 0755 ${WORKDIR}/eth.network ${D}${sysconfdir}/systemd/network
+  fi
+}
```
I find it incredibly odd that there isn't already a standard way to achieve this. But no use in wondering, this is a good fix for now!

kacf commented:
If you fix the not-yet-quite-fixed-comment, you can go ahead and merge this yourself. It looks good otherwise!

pasinskim commented:
Done now. Merging!
